### PR TITLE
[v7r2] scriptdocs

### PR DESCRIPTION
**Pull request by TaykYoku**
This PR provide some #4853 and #4954 changes.
Much of the description of scripts from CommandReference `*.rst`'s are transferred to the `__doc__`'s of a script files.

The main change in `LocalConfiguration.py` and `Script` files, other changes concern the design of script docs mainly.

BEGINRELEASENOTES

*All systems affected:
CHANGE: use `__doc__` for script help message, remove filecfg as argument from usage description

*Core:
CHANGE: when import Script, a help message is formed from `__doc__`, not need to use `Script.setUsageMessage(__doc__)`

*Configuration:
CHANGE: setUsageMessage in LocalConfiguration will parse received str value for building help message

ENDRELEASENOTES

### CODE REVIEW

**Comment by fstagni** [comment]
This looks fine to me, with just a small caveat: for the `dirac-version` and `dirac-platfom` script can you align to what I have done in https://github.com/DIRACGrid/DIRAC/pull/4954 ? To avoid possible conflicts.

File: src/DIRAC/ProductionSystem/scripts/dirac_prod_delete.py [hunk 1]
```diff
@@ -1,7 +1,15 @@
 #!/usr/bin/env python
 
 """
-  Delete a given production
+Delete a given production
+
+Usage::
```
Comment (andresailer) [inline src/DIRAC/ProductionSystem/scripts/dirac_prod_delete.py:6]: Single colons (`Usage:`) should be enough, right? is in lots of places, also Arguments, Examples.

Comment (TaykYoku) [reply src/DIRAC/ProductionSystem/scripts/dirac_prod_delete.py:6]: yes

File: src/DIRAC/ConfigurationSystem/Client/LocalConfiguration.py [hunk 2]
```diff
@@ -86,7 +90,20 @@ def setUsageMessage(self, usageMsg):
     """
     Define message to be display by the showHelp method
     """
-    self.__usageMessage = usageMsg
+    r = r"(.*?)(?:\n(?:Usage|Example|Arguments|Options):+\n|$)"
```
[the remark this answers is not in the document]
Comment (TaykYoku) [reply src/DIRAC/ConfigurationSystem/Client/LocalConfiguration.py]: done

File: src/DIRAC/ProductionSystem/scripts/dirac_prod_delete.py (hunk 1 above)
Comment (TaykYoku) [reply src/DIRAC/ProductionSystem/scripts/dirac_prod_delete.py]: corrected everywhere to the format:
```
"""
Script description

Usage:
  script [options] ...

Example:
  $ script
"""
```

**Review by andresailer — APPROVED** [review]

**Review by atsareg — APPROVED** [review]

**Pull request merged**